homeassistant/components/kef/media_player.py

```diff
@@ -0,0 +1,233 @@
+"""Platform for the KEF Wireless Speakers."""
+
+import datetime
+import logging
+
+from aiokef.aiokef import INPUT_SOURCES, AsyncKefSpeaker
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNKNOWN,
+)
+from homeassistant.helpers import config_validation as cv
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+
+DEFAULT_NAME = "KEF"
+DEFAULT_PORT = 50001
+DEFAULT_MAX_VOLUME = 0.5
+DEFAULT_VOLUME_STEP = 0.05
+DATA_KEF = "kef"
+
+SCAN_INTERVAL = datetime.timedelta(seconds=30)
+PARALLEL_UPDATES = 0
+
+KEF_LS50_SOURCES = sorted(INPUT_SOURCES.keys())
+
+SUPPORT_KEF = (
+    SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_VOLUME_MUTE
+    | SUPPORT_SELECT_SOURCE
+    | SUPPORT_TURN_OFF
+    | SUPPORT_TURN_ON
+)
+
+CONF_MAX_VOLUME = "maximum_volume"
+CONF_VOLUME_STEP = "volume_step"
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MAX_VOLUME, default=DEFAULT_MAX_VOLUME): cv.small_float,
+        vol.Optional(CONF_VOLUME_STEP, default=DEFAULT_VOLUME_STEP): cv.small_float,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```

Comment:
Should I make this an `async` function, even when nothing is awaited?

Reply:
Yes, there is no sync equivalent for the setup.
If you need to call sync methods in here, you'll need to wrap it in a job an add it to the executor.

Reply:
OK, no I/O happens inside `__init__` of `AsyncKefSpeaker`, so I can just leave it as it is, right?

Reply:
Yep.